docs/patterns/messaging-bridge-content.md

```diff
@@ -0,0 +1,88 @@
+
+- In order to meet desired availability standard, the messaging bridge might need to be scaled-out using the [Competing Consumers](../../patterns/ccompeting-consumers.yml) approach.
+
+- Regular message-processing components use a [Retry](../../patterns/retry.yml) pattern to cope with transient failures while the retry counter limit allows them to detect _poison_ messages and remove them from the queue to unblock processing. The bridge might require a different retry policy to prevent falsely indentifying messages as poison in case of the infrastructure failure, possibly involving a [Circuir Breaker](../../patterns/circuit-breaker.yml) to pause forwarding.
+
+## When to use this pattern
+
+Use the Messaging Bridge pattern when you need to:
+
+- Integrate existing systems with minimal need for modification
+- Integrate legacy applications that cannot be changed to use any messaging technology
+- Extend existing on-premises applications with cloud-hosted components
+- Connect geo-distributed systems when Internet connection is not stable
+- Migrate a single distributed system from one messaging infrastructure to another without the need to stop and re-deploy the whole system
+
+This pattern might not be suitable if:
+
+- At least one of the systems involved relies on specific feature of one messaging infrastructure not present in the other
+- Integration is synchronous in nature (the initiating system requires immediate response)
+- Integration has specific functional or non-functional requirements (e.g. security or privacy concerns)
+- The volume of data for the integration exceeds the capacity of the messaging system or makes messaging not a cost-effective solution to the problem
+
+## Example
+
+There is a legacy application written in .NET Framework for managing production site hosted on site. The application is well-structured with separate components communicating via Microsoft Message Queueing (MSMQ). The application works just fine and the company has no intention of re-writing it, but the IT strategy calls for building new software as cloud-native applications in order to optimize the costs and delivery time.
+
+Single the asynchronous queue-based architecture worked well for the company in the past, the new components are going to use the same architectural approach but with modern technology -- Azure ServiceBus. The implementation does not want to introduce new concepts, such as REST/HTTP communication, just to integrate the two sides and would rather take advantage of the existing mechanisms.
```

Comment:
```suggestion
Since the asynchronous queue-based architecture worked well for the company in the past, the new components are going to use the same architectural approach but with modern technology -- Azure Service Bus. The company does not want to introduce new concepts, such as REST/HTTP communication, just to integrate the two sides and would rather take advantage of the existing infrastructure.
```